rioxarray/rioxarray.py

```diff
@@ -375,11 +375,16 @@ def nodata(self):
                     self._obj.attrs.get("fill_value", self._obj.attrs.get("nodata")),
                 ),
             )
+
+        # look in places used by `xarray.open_rasterio`
         if self._nodata is None:
             try:
-                self._nodata = self._obj.attrs["nodatavals"][0]
-            except (KeyError, IndexError):
-                pass
+                self._nodata = self._obj._file_obj.acquire().nodata
+            except AttributeError:
+                try:
+                    self._nodata = self._obj.attrs["nodatavals"][0]
+                except (KeyError, IndexError):
+                    pass
 
         if self._nodata is None:
```

Comment:
I know this hasn't changed in this PR, but are there any concerns with setting nodata to False? What's the benefit over None?

Reply:
Setting it to False make it so that the entire process for looking for the nodata value wont have to be re-run through each time in the case that the value is None. See the beginning of the method.

Reply:
https://github.com/corteva/rioxarray/blob/8605b618c696a24271f8c87ea405a463364f602f/rioxarray/rioxarray.py#L364-L366